packages/compass-utils/src/intercom-track.ts

```diff
@@ -0,0 +1,24 @@
+type WindowWithIntercomGlobals = Window &
+  typeof globalThis & {
+    Intercom?: (...args: any[]) => any;
+    attachEvent?: (...args: any[]) => any;
+  };
+
+export enum IntercomTrackingEvent {
+  submittedNlPrompt = 'submitted-nl-prompt',
```

Comment:
Is this really the first and only event we're tracking in intercom?

Reply:
but I'm pretty noob in this codebase, so perhaps someone else can confirm 😅

Reply:
duplicate conversation in https://mongodb.slack.com/archives/G2L10JAV7/p1707914433425239